tests/integration/test_testpypi_version.py

```diff
@@ -0,0 +1,34 @@
+import unittest
+import requests
+from packaging import version
+# already part of python
+
+try:
+    from .version import git_revision as __git_revision__
+    from .version import version as __version__
+except ModuleNotFoundError:
+    from setup import get_version_info
+    FULLVERSION, GIT_REVISION = get_version_info()
+    __version__ = FULLVERSION
+    __git_revision__ = GIT_REVISION
+
+
+class TestPyPIVersionNumber(unittest.TestCase):
+
+    def test_version_number(self):
+
+        # test for TestPyPI version number
+
+        package = 'macrosynergy'
+        with requests.get(f'https://test.pypi.org/pypi/{package}/json') as r:
```

Comment:
Why not put the `package` name directly in the URL, so that it becomes `with requests.get(f'https://test.pypi.org/pypi/macrosynergy/json') as r:`?